src/raft_log.rs

```diff
@@ -355,11 +355,20 @@ impl<T: Storage> RaftLog<T> {
     }
 
     /// Returns any entries since the a particular index.
-    pub fn next_entries_since(&self, since_idx: u64) -> Option<Vec<Entry>> {
+    pub fn next_entries_since(&self, since_idx: u64, synced_idx: Option<u64>) -> Option<Vec<Entry>> {
         let offset = cmp::max(since_idx + 1, self.first_index());
-        let committed = self.committed;
-        if committed + 1 > offset {
-            match self.slice(offset, committed + 1, None) {
+        let high = match synced_idx {
```

Comment:
I think you can use `cmp::min(self.committed, self.prs[self].matched)`.

Reply:
Yes thanks!